Cli/Basic.lean

```diff
@@ -232,6 +232,17 @@ section Configuration
       | "" => none
       | s  => s.toInt?
 
+  open Lean System in
+  /-- A custom command-line argument parser that allows either relative paths to Lean files,
+  (e.g. `Mathlib/Topology/Basic.lean`) or the module name (e.g. `Mathlib.Topology.Basic`). -/
+  instance : ParseableType Name where
+    name     := "Name"
+    parse? s :=
+      if s.endsWith ".lean" then
+        some <| (s : FilePath).withExtension "" |>.components.foldl Name.mkStr Name.anonymous
+      else
+        String.toName s
```

Comment:
Note that this yields `unreachable!` when given a digit and after that a literal that cannot be parsed to `Nat`, or `Name.num` when given a literal that parses to `Nat`. Both of these should likely yield `Option.none`.

Reply:
Easy enough to handle the `Name.num` result.

It feels like the `unreachable!` in `String.toName` is an upstream bug, not sure what to do about that; I don't really want to re-implement the function here.

Reply:
How about yielding `Option.none` if the first character is a digit? This would cover both cases.

It would also be nice to have some tests for this `ParseableType` instance in `Tests.lean`.

Reply:
Yeah, it looks like an upstream bug given that `String.toName` is also used in lots of user-facing code there. It's probably good enough to handle only the `Name.num` case then and perhaps fix it upstream some other time as well.

Reply:
Ah, it's not actually a bug. These kinds of names are mapped to `Name.anonymous` by `Lean.Syntax.splitNameLitAux`.